src/api/composition/CompositionCollection.js

```diff
@@ -203,50 +219,39 @@ define([
                 this.mutables[keyString] = child;
             }
 
-            this.emit('add', child);
+            this.#emit('add', child);
         }
-    };
-
+    }
     /**
      * Load the domain objects in this composition.
      *
-     * @returns {Promise.<Array.<module:openmct.DomainObject>>} a promise for
+     * @param {any} abortSignal
+     * @returns {Promise.<Array.<DomainObject>>} a promise for
      *          the domain objects in this composition
      * @memberof {module:openmct.CompositionCollection#}
      * @name load
      */
-    CompositionCollection.prototype.load = function (abortSignal) {
-        this.cleanUpMutables();
-
-        return this.provider.load(this.domainObject)
-            .then(function (children) {
-                return Promise.all(children.map((c) => this.publicAPI.objects.get(c, abortSignal)));
-            }.bind(this))
-            .then(function (childObjects) {
-                childObjects.forEach(c => this.add(c, true));
-
-                return childObjects;
-            }.bind(this))
-            .then(function (children) {
-                this.emit('load');
-
-                return children;
-            }.bind(this));
-    };
-
+    async load(abortSignal) {
+        this.#cleanUpMutables();
+        const children = await this.provider.load(this.domainObject);
+        const childObjects = await Promise.all(children.map((c) => this.publicAPI.objects.get(c, abortSignal)));
+        childObjects.forEach(c => this.add(c, true));
+        this.#emit('load');
+
+        return childObjects;
+    }
     /**
      * Remove a domain object from this composition.
      *
      * A call to [load]{@link module:openmct.CompositionCollection#load}
      * must have resolved before using this method.
      *
-     * @param {module:openmct.DomainObject} child the domain object to remove
+     * @param {DomainObject} child the domain object to remove
      * @param {boolean} skipMutate true if the underlying provider should
```

Comment:
Ugh, `skipMutate` is gross and shouldn't exist. It should only _ever_ be used internally, never by external consumers of this API. Maybe we should just remove it from the docs so it doesn't manifest in the TS type?

Reply:
Since this is an ES6 class, TypeScript will infer its existence when generating the *.d.ts files, so it will be there regardless of whether or not we document it (it just will have an `any` type instead of the correct `boolean`). I'll make a note in the docs that this is not for use by public consumers of the API. I'll also leave a `FIXME` noting that we should get rid of this